test/jdk/java/net/httpclient/AuthFilter.java

```diff
@@ -0,0 +1,177 @@
+        test(true);
+    }
+
+    /**
+     *  Fake proxy. Just looks for Proxy-Authorization header
+     *  and returns error if seen. Returns 200 OK if not.
+     *  Does not actually forward the request
+     */
+    static class ProxyServer extends Thread {
+
+        final ServerSocketChannel server;
+        final int port;
+        volatile SocketChannel c;
+
+        ProxyServer() throws IOException {
+            server = ServerSocketChannel.open();
+            server.bind(new InetSocketAddress(0));
+            if (server.getLocalAddress() instanceof InetSocketAddress isa) {
+                port = isa.getPort();
+            } else {
+                port = -1;
+            }
+        }
+
+        int getPort() {
+            return port;
+        }
+
+        static String ok = "HTTP/1.1 200 OK\r\nContent-Length: 0\r\n\r\n";
+        static String notok1 = "HTTP/1.1 500 Internal Server Error\r\nContent-Length: 0\r\n\r\n";
+        static String notok2 = "HTTP/1.1 501 Not Implemented\r\nContent-Length: 0\r\n\r\n";
+
+        static void reply(String msg, Writer writer) throws IOException {
+            writer.write(msg);
+            writer.flush();
+        }
+
+        public void run() {
+            try {
+                c = server.accept();
+                var cs = StandardCharsets.US_ASCII;
+                LineNumberReader reader = new LineNumberReader(Channels.newReader(c, cs));
+                Writer writer = Channels.newWriter(c, cs);
+
+                String line;
+                while ((line=reader.readLine()) != null) {
+                    if (line.indexOf("Proxy-Authorization") != -1) {
+                        reply(notok1, writer);
+                        return;
+                    }
+                    if (line.equals("")) {
+                        // end of headers
+                        reply(ok, writer);
+                        return;
+                    }
+                }
+                reply(notok2, writer);
+            } catch (IOException e) {
+            }
+            try {
+                server.close();
+                c.close();
+            } catch (IOException ee) {}
+        }
+    }
+
+    public static void test(boolean useProxy) throws Exception {
+        HttpServer server = createServer();
+        int port = server.getAddress().getPort();
+        ProxyServer proxy;
+
+        InetSocketAddress proxyAddr=null;
+        String authHdr;
+        if (useProxy) {
+            proxy = new ProxyServer();
+            proxyAddr = new InetSocketAddress("127.0.0.1", proxy.getPort());
+            proxy.start();
+            authHdr = "Proxy-Authorization";
+        } else {
+            authHdr = "Authorization";
+        }
+
+        server.start();
+
+        HttpClient.Builder builder = HttpClient
+                .newBuilder()
+                .authenticator(new Auth());
+
+        if (useProxy) {
+            builder.proxy(ProxySelector.of(proxyAddr));
+        }
```

Comment:
You should probably set NO_PROXY otherwise to avoid the default proxy selector in mac.

Reply:
For the non-proxy case? Good idea.

Regarding the question above about the `-Djdk.http.auth.proxying.disabledSchemes=""` setting. The test fails for both server and proxy auth without this change (and without having to set that property).